[CI] StatsSummaryTests testEqualsAndHashCode failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3999 / ubuntu-1804_platform-support-unix](https://gradle-enterprise.elastic.co/s/ac7yogcoreusi)
- [elasticsearch-pull-request #30737 / part-1](https://gradle-enterprise.elastic.co/s/a7wt3svzmpequ)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.script.StatsSummaryTests.testEqualsAndHashCode" -Dtests.seed=CF497BFD21C388C1 -Dtests.locale=gd-GB -Dtests.timezone=Europe/Zagreb -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testEqualsAndHashCode'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.script.StatsSummaryTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: not <StatsSummary{count=0, sum=0.000000, min=0.000000, average=0.000000, max=0.000000}>
     but: was <StatsSummary{count=0, sum=0.000000, min=0.000000, average=0.000000, max=0.000000}>
```

**Issue Reasons:**
- [main] 2 failures in test testEqualsAndHashCode (0.8% fail rate in 238 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -103,9 +103,6 @@ tests:
 - class: org.elasticsearch.xpack.sql.qa.single_node.JdbcSqlSpecIT
   method: test {case-functions.testUcaseInline3}
   issue: https://github.com/elastic/elasticsearch/issues/112643
-- class: org.elasticsearch.script.StatsSummaryTests
-  method: testEqualsAndHashCode
-  issue: https://github.com/elastic/elasticsearch/issues/112439
 - class: org.elasticsearch.repositories.blobstore.testkit.analyze.HdfsRepositoryAnalysisRestIT
   issue: https://github.com/elastic/elasticsearch/issues/112889
 - class: org.elasticsearch.xpack.sql.qa.security.JdbcSqlSpecIT

--- a/server/src/test/java/org/elasticsearch/script/StatsSummaryTests.java
+++ b/server/src/test/java/org/elasticsearch/script/StatsSummaryTests.java
@@ -76,8 +76,8 @@ public class StatsSummaryTests extends ESTestCase {
         assertThat(stats1.hashCode(), equalTo(stats2.hashCode()));
 
         // Accumulators with different values are not equals
-        randomDoubles(randomIntBetween(0, 20)).forEach(stats1);
-        randomDoubles(randomIntBetween(0, 20)).forEach(stats2);
+        randomDoubles(randomIntBetween(1, 10)).forEach(stats1);
+        randomDoubles(randomIntBetween(11, 20)).forEach(stats2);
         assertThat(stats1, not(equalTo(stats2)));
         assertThat(stats1.hashCode(), not(equalTo(stats2.hashCode())));
     }

--- a/server/src/test/java/org/elasticsearch/script/StatsSummaryTests.java
+++ b/server/src/test/java/org/elasticsearch/script/StatsSummaryTests.java
@@ -75,9 +75,21 @@ public class StatsSummaryTests extends ESTestCase {
         assertThat(stats1, equalTo(stats2));
         assertThat(stats1.hashCode(), equalTo(stats2.hashCode()));
 
+        // Accumulators with same sum, but different counts are not equals
+        stats1.accept(1);
+        stats1.accept(1);
+        stats2.accept(2);
+        assertThat(stats1.getSum(), equalTo(stats2.getSum()));
+        assertThat(stats1.getCount(), not(equalTo(stats2.getCount())));
+        assertThat(stats1, not(equalTo(stats2)));
+        assertThat(stats1.hashCode(), not(equalTo(stats2.hashCode())));
+
         // Accumulators with different values are not equals
-        randomDoubles(randomIntBetween(1, 10)).forEach(stats1);
-        randomDoubles(randomIntBetween(11, 20)).forEach(stats2);
+        stats1.reset();
+        stats2.reset();
+        randomDoubles(randomIntBetween(1, 20)).forEach(stats1.andThen(v -> stats2.accept(v + randomDoubleBetween(0.0, 1.0, false))));
+        assertThat(stats1.getCount(), equalTo(stats2.getCount()));
+        assertThat(stats1.getSum(), not(equalTo(stats2.getSum())));
         assertThat(stats1, not(equalTo(stats2)));
         assertThat(stats1.hashCode(), not(equalTo(stats2.hashCode())));
     }

